features/device_key/TESTS/host_tests/devicekey_reset.py

```diff
@@ -0,0 +1,150 @@
+"""
+Copyright (c) 2018 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import time
+from mbed_host_tests import BaseHostTest
+from mbed_host_tests.host_tests_runner.host_test_default import DefaultTestSelector
+
+DEFAULT_CYCLE_PERIOD      = 1.0
+MSG_VALUE_DUMMY           = '0'
+MSG_KEY_DEVICE_READY      = 'ready'
+MSG_KEY_DEVICE_FINISH     = 'finish'
+MSG_KEY_DEVICE_TEST_STEP1 = 'check_consistency_step1'
+MSG_KEY_DEVICE_TEST_STEP2 = 'check_consistency_step2'
+MSG_KEY_DEVICE_TEST_STEP3 = 'check_consistency_step3'
+MSG_KEY_DEVICE_TEST_STEP4 = 'check_consistency_step4'
+MSG_KEY_SYNC              = '__sync'
+MSG_KEY_TEST_SUITE_ENDED  = 'Test suite ended'
+
+class DeviceKeyResetTest(BaseHostTest):
+    """Test for the DeviceKey driver API.
+    """
+
+    def __init__(self):
+        super(DeviceKeyResetTest, self).__init__()
+        self.reset = False
+        self.finish = False
+        self.suite_ended = False
+        cycle_s = self.get_config_item('program_cycle_s')
+        self.program_cycle_s = cycle_s if cycle_s is not None else DEFAULT_CYCLE_PERIOD
+        self.test_steps_sequence = self.test_steps()
+        # Advance the coroutine to it's first yield statement.
+        self.test_steps_sequence.send(None)
+
+    def setup(self):
+        self.register_callback(MSG_KEY_DEVICE_READY, self.cb_device_ready)
```

Comment:
Not asking to change this. But for your knowledge there is decorator for registering callbacks https://github.com/ARMmbed/htrun/blob/master/mbed_host_tests/host_tests/base_host_test.py#L190